python/cudf/dataframe/dataframe.py

```diff
@@ -453,18 +453,21 @@ def take(self, positions, ignore_index=False):
             out[col] = self[col].take(positions, ignore_index=ignore_index)
         return out
 
-    def copy(self):
+    def copy(self, deep=True):
         """
         Returns a copy of this dataframe
         """
         df = DataFrame()
         df._index = self._index
         df._size = self._size
-        df._cols = self._cols.copy()
+        if deep:
+            df._cols = self._cols.copy()
```

Comment:
This will make a shallow copy of the OrderedDict and the columns will still be the same underneath. We need to deepcopy the columns as well.